celest/schedule/insertion_operators.py

```diff
@@ -0,0 +1,128 @@
+
+
+from celest.units.quantity import Quantity
+from celest import units as u
+import math
+import numpy as np
+
+
+def greedy_insertion(request_handler, number_to_insert):
+    request_handler.sort_by_decreasing_priority()
+    insert_first_n_requests(request_handler, number_to_insert)
+
+
+def insert_first_n_requests(request_handler, number_to_insert):
+    for request_index in range(request_handler.number_of_requests):
+
+        if not number_to_insert:
+            break
+
+        if request_handler.is_request_scheduled(request_index):
+            continue
+
+        deadline = request_handler.deadline(request_index)
+        duration = request_handler.duration(request_index)
+        minimum_image_quality = request_handler.image_quality(request_index)
+        look_angle = request_handler.look_angle(request_index)
+        request_vtw_list = request_handler.vtw_list(request_index)
+
+        duration_in_days = Quantity(duration.to(u.s) / 86400, u.jd2000)
+
+        for vtw_index, vtw in enumerate(request_vtw_list):
+
+            if look_angle is not None:
+                start = _look_angle_time(
+                    vtw.attitude.pitch,
+                    look_angle,
+                    vtw.attitude.time,
+                    vtw.rise_time,
+                    vtw.set_time
+                )
+                if start is None:
+                    continue
+            else:
+                start = (vtw.rise_time + vtw.set_time + duration_in_days) / 2
+
+            if (start < vtw.rise_time) or (start + duration_in_days > vtw.set_time):
+                continue
+            if start + duration_in_days > deadline:
+                continue
+            if not image_quality_is_met(vtw, start, minimum_image_quality):
+                continue
+            if does_conflict_exists(request_handler, start, duration_in_days):
+                continue
+
+            request_handler.schedule_request(request_index, vtw_index, start, duration)
+            number_to_insert -= 1
+            break
+
+
+def _look_angle_time(look_angle, desired_look_angle, julian, start, end):
```

Comment:
This is a very messy implementation of a very simple algorithm. Perhaps, extract a bisection algorithm function and call it from here.